api/tests/integration/infrastructure/repositories/badge-acquisition-repository_test.js

```diff
@@ -29,26 +26,16 @@ describe('Integration | Repository | Badge Acquisition', () => {
 
     it('should persist the badge acquisition in db', async () => {
       // when
-      badgeAcquisition = await DomainTransaction.execute(async (domainTransaction) => {
-        return badgeAcquisitionRepository.create(badgeAcquisitionToCreate, domainTransaction);
+      const badgeAcquisitionIds = await DomainTransaction.execute(async () => {
+        return badgeAcquisitionRepository.create([badgeAcquisitionToCreate]);
       });
 
       // then
-      const result = await knex('badge-acquisitions').where('id', badgeAcquisition.id);
-
+      expect(badgeAcquisitionIds).to.have.lengthOf(1);
+      const result = await knex('badge-acquisitions').where('id', badgeAcquisitionIds[0]);
       expect(result).to.have.lengthOf(1);
     });
 
-    it('should return the saved badge acquired', async () => {
-      // when
-      badgeAcquisition = await DomainTransaction.execute(async (domainTransaction) => {
-        return badgeAcquisitionRepository.create(badgeAcquisitionToCreate, domainTransaction);
-      });
-
-      // then
-      expect(badgeAcquisition).to.be.an.instanceOf(BadgeAcquisition);
-      expect(badgeAcquisition).to.have.property('id').and.not.to.be.null;
-    });
```

Comment:
Pourquoi on ne teste plus le résultat de la transaction ?

Reply:
On teste déjà si le badge est créé dans le test précédent.
Ce test ci teste si on a bien un objet `badgeAcquisition` en sortie de la fonction, or il n'y en a pas besoin.

Reply:
Est-ce que c'est toujours d'actualité avec le retour de la DomainTransaction ? En général, je n'aime pas trop enlever des tests

Reply:
Le test `should return the saved badge acquired` est supprimé parce que la création des badges ne retourne désormais plus de badges, mais les id des badges créés. 
Le test `should persist the badge acquisition in db` vérifie bien que des ids sont retournés après la création de badges. Ce test est d'actualité avec le retour de DomainTransaction.